pkg/block/index.go

```diff
@@ -347,9 +407,50 @@ func Repair(dir string, id ulid.ULID) (resid ulid.ULID, err error) {
 
 var castagnoli = crc32.MakeTable(crc32.Castagnoli)
 
+func IgnoreCompleteOutsideChunk(mint, maxt int64, _ *chunks.Meta, curr *chunks.Meta) (bool, error) {
+	if curr.MinTime > maxt || curr.MaxTime < mint {
+		// "Complete" outsider. Ignore.
+		return true, nil
+	}
+	return false, nil
+}
+
+func IgnoreIssue347OutsideChunk(mint, maxt int64, _ *chunks.Meta, curr *chunks.Meta) (bool, error) {
+	if curr.MinTime == maxt && curr.MaxTime > maxt {
+		// "Near" outsider from issue https://github.com/prometheus/tsdb/issues/347. Ignore.
+		return true, nil
+	}
+	return false, nil
+}
```

Comment:
I can see why you would collect separate stats for outside chunks and complete outside chunks, but when it comes to repairing blocks, they're really the same thing.

I mean if you have two intervals, `block = [a, b)` and `chunk = [c, d]`, they don't overlap at all even if `b == c`. So you could have a single ignore function:

```go
func IgnoreOutsideChunk(mint, maxt int64, _ *chunks.Meta, curr *chunks.Meta) (bool, error) {
	if curr.MinTime >= maxt || curr.MaxTime < mint {
		return true, nil
	}
	return false, nil
}
```

If you don't merge those two functions, then I think you should change the condition in `IgnoreIssue347OutsideChunk()` to 

```go
if curr.MinTime == maxt {
	return true, nil
}
```

Reply:
Yea will do latter. Can agree that "complete" and "near" is kind of the same -> we can kill them just fine, but there is no upstream issue causing "complete". with issue347 we know what is that root cause and we know that it will not  be fixed until it lands and land into Prometheus AND Prometheus go released. This will be quite a long time. With `complete` and `duplicates` even though they safe to repair they only occurred once Thanos had an issue, so the only explicit manual command should be fixing these, nothing automated.